src/mame/sinclair/sprinter.cpp

```diff
@@ -0,0 +1,1751 @@
+	case 0x13:
+		m_beta->data_w(data);
+		break;
+	case 0x14:
+		m_beta->param_w(data);
+		break;
+	case 0x16:
+		if (data & 2)
+			m_beta->disable();
+		else
+			m_beta->enable();
+		break;
+
+	case 0x1d:
+	case 0x1e:
+		m_rtc->write(~dcpp & 1, data);
+		break;
+
+	case 0x20:
+		if (BIT(offset, 8))
+			m_ata[m_ata_selected]->cs0_w(0, (data << 8) | m_ata_data_latch);
+		else
+			m_ata_data_latch = data;
+		break;
+	case 0x21 ... 0x27:
+		if (BIT(offset, 8))
+			m_ata[m_ata_selected]->cs0_w(dcpp & 0x07, data);
+		break;
+	case 0x28:
+		if (BIT(offset, 8))
+			m_ata[m_ata_selected]->cs1_w(6, data);
+		break;
+	case 0x2a: // HDD1 - secondary
+		m_ata_selected = 1;
+		break;
+	case 0x2b: // HDD2 - primary
+		m_ata_selected = 0;
+		break;
+	case 0x2e:
+		if (m_conf)
+			machine().schedule_hard_reset();
+		else
+		{
+			m_conf_loading = 1;
+			machine().schedule_soft_reset();
+		}
+		break;
+
+	case 0x88:
+		if (CBL_MODE)
+			m_cbl_data[m_cbl_wa++] = data << 8;
+		else
+		{
+			m_ldac->write(data << 8);
+			m_rdac->write(data << 8);
+		}
+		break;
+
+	case 0x89:
+	{
+		m_cbl_xx = data;
+		m_cbl_cnt = 0;
+		m_cbl_wa = 0;
+		m_cbl_wae = CBL_MODE16;
+		const u8 divs[16] = {13, 9, 0, 0, 0, 0, 0, 0, 27, 19, 13, 9, 6, 4, 3, 1};
+		const attotime rate = (CBL_MODE && divs[m_cbl_xx & 15]) ? attotime::from_ticks(divs[m_cbl_xx & 15] + 1, X_SP / 192) : attotime::never;
+		m_cbl_timer->adjust(rate, 0, rate);
+		break;
+	}
+
+	case 0x8f: // rom & fastram page
+		m_rom_rg = data;
+		m_sys_pg = BIT(m_rom_rg, 4) || BIT(data, 0);
+		update_memory();
+		break;
+
+	case 0x90: // AY8910
+		m_ay8910->address_w(data);
+		break;
+	case 0x91:
+		m_ay8910->data_w(data);
+		break;
+
+	case 0xc0: // 1FFD
+	case 0xc8:
+		m_sc = data;
+		if (BIT(m_cnf, 6)) m_sc = 0;      // CNF_SC_RESET
+		update_memory();
+		break;
+	case 0xc1: // 7FFD
+	case 0xc9:
+		m_pn = data;
+		if (BIT(m_cnf, 5)) m_pn &= 0xc0;  // CNF_PN[5..0]_RESET
+		if (BIT(~m_cnf, 7)) m_pn &= 0x1f; // CNF_PN[7..6]_RESET
+		update_memory();
+		break;
+	case 0xc2:
+		m_screen->update_now();
+		spectrum_ula_w(offset, data);
+		break;
+	case 0xc3:
+		m_all_mode = data;
+		break;
+	case 0xcb:
+		m_hold = data;
+		break;
+	case 0xc4:
+	case 0xcc:
+		m_port_y = data;
+		break;
+	case 0xc5:
+	case 0xcd:
+		m_screen->update_now();
+		m_rgmod = data;
+		update_int(true);
+		break;
+
+	case 0xc6: // CNF/SYS
+	case 0xce:
+		m_ram_sys = BIT(~offset, 6);
+		if (BIT(data, 1))
+		{
+			m_turbo = BIT(data, 0);
+			update_cpu();
+		}
+		else
+			m_arom16 = BIT(data, 0);
+
+		if (BIT(data, 2))
+		{
+			m_cnf = data;
+			if (BIT(m_cnf, 5)) m_pn &= 0xc0;  // CNF_PN[5..0]_RESET
+			if (BIT(m_cnf, 6)) m_sc = 0;      // CNF_SC_RESET
+			if (BIT(~m_cnf, 7)) m_pn &= 0x1f; // CNF_PN[7..6]_RESET
+		}
+
+		update_memory();
+		break;
+
+	case 0xc7:
+	case 0xcf:
+		m_alt_acc = 1;
+		m_aagr = (offset & 0x300) | data;
+		m_xcnt = BIT(offset, 10, 6);
+		m_xagr = 0;
+		break;
+
+	case 0xd0 ... 0xef:
+		update_memory();
+		break;
+	case 0xf0 ... 0xff:
```

Comment:
Unfortunately case ranges are a proprietary extension and I’m going to have to ask you to refrain from using them.